src/ElfUtils/ElfFile.cpp

```diff
@@ -385,4 +389,33 @@ ErrorMessageOr<std::unique_ptr<ElfFile>> ElfFile::Create(
   return result;
 }
 
+ErrorMessageOr<uint32_t> ElfFile::CalculateDebuglinkChecksum(
+    const std::filesystem::path& file_path) {
+  // TODO(b/180995172): Make this read operation iterative since the potentially read files can be
+  // very large (gigabytes).
+  ErrorMessageOr<orbit_base::unique_fd> fd_or_error = orbit_base::OpenFileForReading(file_path);
+
+  if (fd_or_error.has_error()) {
+    return fd_or_error.error();
+  }
+
+  constexpr size_t kBufferSize = 4 * 1024 * 1024;  // 4 MiB
+  std::vector<char> buffer(kBufferSize);
+
+  uint32_t rolling_checksum = 0;
+
+  while (true) {
+    ErrorMessageOr<size_t> chunksize_or_error =
+        orbit_base::ReadFully(fd_or_error.value(), buffer.data(), buffer.size());
+    if (chunksize_or_error.has_error()) return chunksize_or_error.error();
+
+    const llvm::StringRef str{buffer.data(), chunksize_or_error.value()};
```

Comment:
chunksize_or_error could be 0 in case file size is divisible by 4MiB (not sure if StringRef handles this correctly), maybe just check for 0 and break here instead of doing the check on L415. We can afford extra read once we hit the end of file.

Reply:
done.